### PR TITLE
Fix false positive in SpringHandler for SpEL expressions using null in comparisons

### PR DESCRIPTION
**Pull request by Beluomini**
## What and why

This PR fixes a false positive in `SpringHandler` for Spring `@Value` fields whose SpEL expression
mentions `null` only as a comparison operand.

### Bug fix

**`SpringHandler`: false positive for SpEL expressions with `null` in comparisons**

`containsNullSpELExpression` used a single regex that matched `null` anywhere inside a `#{...}`
expression. This caused a false positive for conditional expressions like:

```java
@Value("#{someBean != null ? someBean.value : 'default'}")
private String field;
```

NullAway incorrectly flagged `field` as potentially null, even though `null` here is only used in
a comparison, not as a possible return value. Fix: a second regex (`NULL_COMPARISON_PATTERN`) strips
null-in-comparison occurrences before re-checking whether `null` can actually be the value.

The heuristic still (correctly) treats the field as possibly uninitialized when `null` can actually
be produced by the expression, e.g. `#{someBean != null ? null : 'default'}`.

### New tests

- **`FrameworkTests.springValueSpelWithNullInConditional`** — regression test for the case above:
  `null` used only as a comparison operand no longer triggers an uninitialized-field error.

- **`FrameworkTests.springValueSpelNullAsReturnValue`** — boundary case where `null` appears both in
  a comparison *and* as a ternary return value (then-branch and else-branch); the error is still
  reported.

- **`FrameworkTests.springValueSpelNullComparisonLeftSide`** — symmetric case with `null` on the left
  side of `!=` / `==`.

### Unit tests

Yes. The first commit adds the test demonstrating the false positive; the second commit contains the
fix together with the additional boundary-case tests.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved Spring `@Value` null detection for SpEL expressions.
  - Equality comparisons involving `null` no longer trigger incorrect non-null initialization diagnostics.
  - Continued detecting expressions that may return `null`, including ternary expressions and comparisons with `null` on either side.

- **Tests**
  - Added coverage for Spring `@Value` SpEL expressions with comparison-only `null` values.
  - Added coverage for nullable ternary return branches and left-side `null` comparisons.

### CODE REVIEW

**Comment by msridhar** [comment]
@Beluomini thanks for the PRs!  Before we can review we'll need you to sign the CLA.

File: nullaway/src/main/java/com/uber/nullaway/handlers/SpringHandler.java [hunk 1]
```diff
@@ -20,12 +20,21 @@ public class SpringHandler implements Handler {
...
    * token. This lets us distinguish Spring {@code @Value} expressions that may produce {@code null}
-   * from plain property placeholders or string literals containing the letters {@code null}. This
-   * is a heuristic match and may have false positives.
+   * from plain property placeholders or string literals containing the letters {@code null}.
    */
   private static final Pattern VALUE_NULL_SPEL_PATTERN =
       Pattern.compile("#\\{[^}]*\\bnull\\b[^}]*}");
 
+  /**
+   * Matches {@code null} used as an operand in equality comparisons ({@code null == null}, {@code
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/handlers/SpringHandler.java:29]: Why are we matching for `null == null` here?  That seems like it would be very uncommon?  If there's no good reason to write that, we can simplify this regex and just not match for it

File: nullaway/src/test/java/com/uber/nullaway/FrameworkTests.java [hunk 2]
```diff
@@ -3201,4 +3205,83 @@ void testPositiveDoOnNextWithoutFilter(Mono<Foo> mono) {
...
+  public void springValueSpelNullComparedToNull() {
+    addSpringValueAnnotationStub(defaultCompilationHelper)
+        .addSourceLines(
+            "Test.java",
+            """
+            package com.uber;
+            import org.springframework.beans.factory.annotation.Value;
+            class Test {
+              // Boundary: null on both sides of the comparison; the whole comparison is stripped.
+              @Value("#{null == null ? 'a' : 'b'}")
```
Comment (msridhar) [inline nullaway/src/test/java/com/uber/nullaway/FrameworkTests.java:3279]: If we don't need to support this case, we can remove this test as well

**Review by msridhar — COMMENTED** [review]
Thanks.  One concern

**Comment by msridhar** [comment]
@Beluomini will you be able to look at the review comments on this one?  Just checking in

File: nullaway/src/main/java/com/uber/nullaway/handlers/SpringHandler.java (hunk 1 above)
Comment (Beluomini) [reply nullaway/src/main/java/com/uber/nullaway/handlers/SpringHandler.java]: Agreed, removed springValueSpelNullComparedToNull since we no longer special-case null == null / null != null.

File: nullaway/src/test/java/com/uber/nullaway/FrameworkTests.java (hunk 2 above)
Comment (Beluomini) [reply nullaway/src/test/java/com/uber/nullaway/FrameworkTests.java]: You're right, null == null isn't a realistic pattern in Spring @Value SpEL expressions. Simplified NULL_COMPARISON_PATTERN to only match null as an operand of ==/!= (removed the two-null alternative) and updated the Javadoc.

**Review by msridhar — APPROVED** [review]
Thanks!
